src/ppom-middleware.test.ts

```diff
@@ -0,0 +1,89 @@
+import { storageBackendReturningData } from '../test/test-utils';
+import { PPOMController } from './ppom-controller';
+import { createPPOMMiddleware } from './ppom-middleware';
+
+jest.mock('./ppom.ts', () => ({
+  PPOM: class PPOMClass {
+    #jsonRpcRequest;
+
+    constructor(jsonRpcRequest: any) {
+      this.#jsonRpcRequest = jsonRpcRequest;
+    }
+
+    validateJsonRpc = async () => {
+      return Promise.resolve();
+    };
+
+    free = () => undefined;
+
+    testJsonRPCRequest = () => this.#jsonRpcRequest();
+  },
+  ppomInit: () => undefined,
+}));
+
+describe('createPPOMMiddleware', () => {
+  it('should return PPOM Middleware function', () => {
+    const ppomController = new PPOMController({
+      storageBackend: storageBackendReturningData,
+      provider: { sendAsync: Promise.resolve() },
+      chainId: '0x1',
+      onNetworkChange: (_callback) => undefined,
+    });
+    const middlewareFunction = createPPOMMiddleware(ppomController);
+    expect(middlewareFunction).toBeDefined();
+  });
+
+  describe('PPOMMiddleware', () => {
+    it('should call ppomController.use when invoked', async () => {
+      const useMock = jest.fn();
+      const controller = {
+        use: useMock,
+      };
+      const middlewareFunction = createPPOMMiddleware(controller as any);
+      await middlewareFunction({}, undefined, () => undefined);
+      expect(useMock).toHaveBeenCalledTimes(1);
```

Comment:
Do we need a test to verify the `ppomResponse` property is set on the original request?

Reply:
Yep that is good idea, I will add the test.